azurerm/resource_arm_api_management.go

```diff
@@ -0,0 +1,773 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2017-03-01/apimanagement"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApiManagementService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApiManagementServiceCreateUpdate,
+		Read:   resourceArmApiManagementServiceRead,
+		Update: resourceArmApiManagementServiceCreateUpdate,
+		Delete: resourceArmApiManagementDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateApiManagementName,
```

Comment:
Could we add this to the azure package in a file api_management.go? and then use it for the data source too